source/MaterialXGenShader/ShaderGenerator.h

```diff
@@ -108,6 +108,17 @@ class ShaderGenerator
         return _sourceCodeSearchPath;
     }
 
+    /// Given a element attempt to remap a value to an enumeration which is accepted by
+    /// the shader generator.
+    /// @param value Value element input
+    /// @param element Element to test
+    /// @param enumType Enumeration type (returned).
+    /// @return Enumeration value. Null if no remapping is performed
+    virtual ValuePtr remapEnumerationValue(const ValueElementPtr& /*value*/, const InterfaceElement& /*element*/, string& /*enumType*/)
```

Comment:
Maybe consider renaming the parameters for clarity:
value -> input
element -> node (or parent or similar)

Also the overridden method on GlslShaderGenerator above should use the same naming.

Reply:
Could the enumType returned be a TypeDesc* instead of a string?

Reply:
I'll look at the renaming. I guess I could but return a reference to a TypeDesc* which I originally had but this seemed easier to return.